frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -2037,6 +2038,122 @@ def encrypt(self, resulting_type: ConversionType, x: Conversion) -> Conversion:
     def equal(self, resulting_type: ConversionType, x: Conversion, y: Conversion) -> Conversion:
         return self.comparison(resulting_type, x, y, accept={Comparison.EQUAL})
 
+    def extract_bit(
+        self,
+        resulting_type: ConversionType,
+        x: Conversion,
+        bit: Union[int, np.integer],
+    ) -> Conversion:
+        if bit >= x.original_bit_width:
+            return (
+                self.zeros(resulting_type)
+                if x.is_unsigned
+                else self.extract_bit(resulting_type, x, x.bit_width - 1)
+            )
+
+        for _ in range(bit):
+            lsb = self.lsb(x.type, x)
+            cleared = self.sub(x.type, x, lsb)
+            x = self.reinterpret(cleared, bit_width=(x.bit_width - 1))
+
+        result = self.to_signedness(self.lsb(x.type, x), of=resulting_type)
+
+        if result.bit_width > resulting_type.bit_width:
+            difference = result.bit_width - resulting_type.bit_width
+            shifter = self.constant(self.i(result.bit_width + 1), 2**difference)
+            shifted = self.mul(result.type, result, shifter)
+            result = self.reinterpret(shifted, bit_width=resulting_type.bit_width)
+
+        elif result.bit_width < resulting_type.bit_width:
+            shifter = self.constant(self.i(result.bit_width + 1), 2 ** (result.bit_width - 1))
+            shifted = self.mul(result.type, result, shifter)
+            reinterpreted = self.reinterpret(shifted, bit_width=1)
+            result = self.tlu(resulting_type, reinterpreted, [0, 1])
```

Comment:
That could be done in the initial lsb, asking directly target precision. (i.e. the max between the resulting precision and the original precision).
It would be less costly.

Reply:
You mean I can do:
```
    %0 = "FHE.lsb"(%arg0) : (!FHE.eint<4>) -> !FHE.eint<7>
```
:thinking:

Reply:
Done.